pkg/relayer/cmd/cmd_relay.go

```diff
@@ -336,90 +338,115 @@ func runRelay(cmd *cobra.Command, args []string) error {
 	}
 	logger.Info().Msgf("✅ Endpoint URL parsed: %v", reqUrl)
 
-	// Create the HTTP request with the relay request body
-	httpReq := &http.Request{
-		Method: http.MethodPost,
-		URL:    reqUrl,
-		Body:   io.NopCloser(bytes.NewReader(relayReqBz)),
-	}
+	// Send multiple requests as specified by the count flag
+	for i := 1; i <= flagRelayRequestCount; i++ {
+		if flagRelayRequestCount > 1 {
+			logger.Info().Msgf("📤 Sending request %d of %d", i, flagRelayRequestCount)
+		}
 
-	// Send the request HTTP request containing the signed relay request
-	httpResp, err := http.DefaultClient.Do(httpReq)
-	if err != nil {
-		logger.Error().Err(err).Msg("❌ Error sending relay request")
-		return err
-	}
-	defer httpResp.Body.Close()
+		// Create the HTTP request with the relay request body
+		httpReq := &http.Request{
+			Method: http.MethodPost,
+			URL:    reqUrl,
+			Body:   io.NopCloser(bytes.NewReader(relayReqBz)),
+		}
 
-	// Read the response
-	respBz, err := io.ReadAll(httpResp.Body)
-	if err != nil {
-		logger.Error().Err(err).Msg("❌ Error reading response")
-		return err
-	}
-	logger.Info().Msgf("✅ Response read %d bytes", len(respBz))
+		// Send the request HTTP request containing the signed relay request
+		httpResp, err := http.DefaultClient.Do(httpReq)
+		if err != nil {
+			logger.Error().Err(err).Msgf("❌ Error sending relay request %d", i)
+			return err
+		}
+		defer httpResp.Body.Close()
```

Comment:
`defer httpResp.Body.Close()` could introduce memory leaks, performance issues and bad connection management for high `flagRelayRequestCount` values because it is inside the `for i := 1; i <= flagRelayRequestCount; i++`. I would recommend explicitly closing the connection right after the next instruction `io.ReadAll()`

Reply:
618c41c1218db6acd647dbfa11e8e9c44a8f7cab